foursight_core/react/api/react_api.py

```diff
@@ -0,0 +1,447 @@
+            environment_and_bucket_info = None
+            portal_url = None
+        # Get known envs with GAC name for each.
+        response = self.create_standard_response("reactapi_info")
+        response.body = {
+            "app": {
+                "title": app.core.html_main_title,
+                "package": app.core.APP_PACKAGE_NAME,
+                "stage": stage_name,
+                "version": app.core.get_app_version(),
+                "domain": domain,
+                "context": context,
+                "local": app.core.is_running_locally(request),
+                "credentials": self.auth.get_aws_credentials(env if env else default_env),
+                "launched": app.core.init_load_time,
+                "deployed": app.core.get_lambda_last_modified()
+            },
+            "versions": {
+                "foursight": app.core.get_app_version(),
+                "foursight_core": pkg_resources.get_distribution('foursight-core').version,
+                "dcicutils": pkg_resources.get_distribution('dcicutils').version,
+                "python": platform.python_version(),
+                "chalice": chalice_version
+            },
+            "server": {
+                "foursight": socket.gethostname(),
+                "portal": portal_url,
+                "es": app.core.host,
+                "rds": os.environ["RDS_HOSTNAME"],
+                # TODO: cache this (slow).
+                "sqs": app.core.sqs.get_sqs_queue().url,
+            },
+            # TODO: cache this (slow).
+            "buckets": {
+                "env": app.core.environment.get_env_bucket_name(),
+                "foursight": get_foursight_bucket(envname=env if env else default_env, stage=stage_name),
+                "foursight_prefix": get_foursight_bucket_prefix(),
+                "info": environment_and_bucket_info,
+                "ecosystem": sort_dictionary_by_lowercase_keys(EnvUtils.declared_data()),
+            },
+            "page": {
+                "path": request.get("context").get("path"),
+                "endpoint": request.get("path"),
+                "loaded": app.core.get_load_time()
+            },
+            # TODD: Move these out to another API.
+            "checks": {
+                "running": 0,
+                "queued": 0
+            },
+            "known_envs": self.envs.get_known_envs_with_gac_names(self.gac),
+            # TODO: cache this (slow).
+            "gac": {
+                "name": get_identity_name(),
+                "values": sort_dictionary_by_lowercase_keys(obfuscate_dict(get_identity_secrets())),
+             },
+            "environ": sort_dictionary_by_lowercase_keys(obfuscate_dict(dict(os.environ)))
+        }
+        return response
+
+    def reactapi_users(self, request: dict, env: str):
+        users = []
+        # TODO: Support paging.
+        user_records = ff_utils.get_metadata('users/', ff_env=full_env_name(env), add_on='frame=object&limit=10000')
```

Comment:
You might consider adding the ability to search for both normal users and admin/foursight users (who would have access to foursight). More advanced: the ability to grant `foursight` access.